File: CODIGOS5/173.dart
import 'dart:io';

void main() {
  int numTestCases = int.parse(stdin.readLineSync()!);

  for (int i = 0; i < numTestCases; i++) {
    List<String> inputNotes = stdin.readLineSync()!.split(' ');
    List<int> notes = inputNotes.map(int.parse).toList();

    String chordName = determineChordName(notes);
    stdout.write('$chordName ');
  }
}

String determineChordName(List<int> notes) {
  List<int> steps = [];
  for (int i = 1; i < notes.length; i++) {
    int diff = (notes[i] - notes[i - 1] + 12) % 12;
    steps.add(diff);
  }

  if (steps.contains(3) && steps.contains(4)) {
    return 'other';
  } else if (steps.contains(4)) {
    return getChordName(notes[0], true);
  } else if (steps.contains(3)) {
    return getChordName(notes[0], false);
  } else {
    return 'other';
  }
}

String getChordName(int rootNote, bool isMajor) {
  List<String> noteNames = [
    'C', 'C#', 'D', 'D#', 'E', 'F', 'F#', 'G', 'G#', 'A', 'A#', 'B'
  ];
  String rootNoteName = noteNames[rootNote % 12];
  return '$rootNoteName ${isMajor ? "major" : "minor"}';
}

File: CODIGOS5/173.scala
object ChordsOfMusic {
  def determineChordName(notes: List[Int]): String = {
    val steps = notes.sliding(2).map { case List(a, b) => (b - a + 12) % 12 }.toList

    if (steps.contains(3) && steps.contains(4)) {
      "other"
    } else if (steps.contains(4)) {
      getChordName(notes.head, isMajor = true)
    } else if (steps.contains(3)) {
      getChordName(notes.head, isMajor = false)
    } else {
      "other"
    }
  }

  def getChordName(rootNote: Int, isMajor: Boolean): String = {
    val noteNames = List("C", "C#", "D", "D#", "E", "F", "F#", "G", "G#", "A", "A#", "B")
    val rootNoteName = noteNames(rootNote % 12)
    if (isMajor) {
      s"${rootNoteName}-major"
    } else {
      s"${rootNoteName}-minor"
    }
  }

  def main(args: Array[String]): Unit = {
    val numTestCases = scala.io.StdIn.readInt()

    for (_ <- 1 to numTestCases) {
      val inputNotes = scala.io.StdIn.readLine().split(" ").map(_.toInt).toList
      val chordName = determineChordName(inputNotes)
      print(s"$chordName ")
    }
  }
}
